<launch>

  <!--rviz中的fixed frame-->
  <arg name="fixed_frame" default="velodyne" />
  <!--裁剪近处自车点x范围-->
  <arg name="cut_x" default="1.5" />
  <!--裁剪近处自车点y范围-->
  <arg name="cut_y" default="0.8" />
  <!--radar x方向偏移量-->
  <arg name="x_offset" default="1.6" />
  <!--lidar ROI width-->
  <arg name="ROI_width" default="5.5" />
  <!--lidar ROI length-->
  <arg name="ROI_length" default="23.0" />

  <!--连接VLP16-->
  <include file="$(find velodyne_pointcloud)/launch/VLP16_points.launch" />

  <!--接收MPC单片机CAN原始数据-->
  <node pkg="socketcan_bridge" type="socketcan_to_topic_node" name="socketcan_to_topic_node" />
  <node pkg="raw_data" type="MPC_raw_node" name="MPC_raw_node" output="screen">
    <param name="fixed_frame" value="$(arg fixed_frame)" />
    <param name="x_offset" value="$(arg x_offset)" />
  </node>  

  <!--激光雷达目标点云聚类-->
  <node pkg="detection" type="lidar_cluster_node" name="lidar_cluster_node" output="screen">
    <param name="fixed_frame" value="$(arg fixed_frame)" />
    <param name="ROI_width" value="$(arg ROI_width)" />
    <param name="ROI_length" value="$(arg ROI_length)" />
    <param name="cut_x" value="$(arg cut_x)" />
    <param name="cut_y" value="$(arg cut_y)" />
  </node>

  <!--目标跟踪模块-->
  <node pkg="detection" type="tracker_node" name="tracker_node" output="screen">
    <param name="fixed_frame" value="$(arg fixed_frame)" />
    <param name="x_offset" value="$(arg x_offset)" />
  </node>

  <!--rviz可视化-->
  <node pkg="rviz" type="rviz" name="rviz" args="-d $(find detection)/rviz/detection.rviz" required="true"/>   

  <!--rqt_plot>
  <node pkg="rqt_plot" type="rqt_plot" name="rqt_plot" /-->

</launch>